packages/compose/src/hooks/use-dragging/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useCallback, useState, useRef } from '@wordpress/element';
+
+export default function useDragging( { onDragStart, onDragMove, onDragEnd } ) {
+	const [ isDragging, setIsDragging ] = useState( false );
+
+	const eventsRef = useRef( {
+		onDragStart,
+		onDragMove,
+		onDragEnd,
+	} );
+	useEffect(
+		() => {
+			eventsRef.current.onDragStart = onDragStart;
+			eventsRef.current.onDragMove = onDragMove;
+			eventsRef.current.onDragEnd = onDragEnd;
+		},
+		[ onDragStart, onDragMove, onDragEnd ]
+	);
+
+	const startDrag = useCallback( ( ...args ) => {
+		if ( eventsRef.current.onDragEnd ) {
+			eventsRef.current.onDragStart( ...args );
+		}
+		setIsDragging( true );
+	} );
```

Comment:
Missing deps.

Reply:
Hi @epiqueras, In this case, the dependencies are a state setter and a reference both of them are never expected to change.
I remember in a previous PR you referring that there is no need to include setter in the dependency. Should we just include the reference in this case?

Reply:
You still need an empty array, otherwise, there is no memoization or reason to use `useCallback`.